fennel/datasets/datasets.py

```diff
@@ -764,19 +777,43 @@ def __init__(
         node: _Node,
         keys: List[str],
         type: WindowType,
-        gap: Duration,
         field: str,
+        gap: Optional[Duration],
+        duration: Optional[Duration],
+        stride: Optional[Duration],
     ):
         super().__init__()
         if len(keys) == 0:
             raise ValueError(
                 "'group_by' before 'window' must specify at least one key"
             )
+        if type == WindowType.Hopping:
```

Comment:
Could you please add negative checks too i.e. if type is Hopping, gap isn't provided. Or if it's tumbling or session, stride isn't provided.